File: CQRS-write/cqrswrite-infrastructure/src/main/java/com/example/cqrswrite/domain/feed/persistence/mapper/FeedMapper.java
package com.example.cqrswrite.domain.feed.persistence.mapper;

import com.example.cqrswrite.domain.feed.model.Feed;
import com.example.cqrswrite.domain.feed.persistence.entity.FeedEntity;
import com.example.cqrswrite.domain.user.exception.UserException;
import com.example.cqrswrite.domain.user.persistence.entity.UserEntity;
import com.example.cqrswrite.domain.user.persistence.repository.UserJpaRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class FeedMapper {

    private final UserJpaRepository userJpaRepository;

    public Feed toDomain(FeedEntity entity) {
        return Feed.builder()
                .id(entity.getId())
                .title(entity.getTitle())
                .content(entity.getContent())
                .userId(entity.getUserEntity().getId())
                .createdAt(entity.getCreatedAt())
                .build();
    }

    public FeedEntity toEntity(Feed domain) {
        UserEntity userEntity = userJpaRepository.findById(domain.getUserId())
                .orElseThrow(() -> UserException.USER_NOT_FOUND);

        return FeedEntity.builder()
                .id(domain.getId())
                .title(domain.getTitle())
                .content(domain.getContent())
                .userEntity(userEntity)
                .createdAt(domain.getCreatedAt())
                .build();
    }
}

File: CQRS-read/cqrsread-infrastructure/src/main/kotlin/com/example/cqrsread/domain/feed/mapper/FeedMapper.kt
package com.example.cqrsread.domain.feed.mapper

import com.example.cqrsread.domain.feed.model.FeedDetails
import com.example.cqrsread.domain.feed.model.Feeds
import com.example.cqrsread.domain.feed.persistence.entity.FeedDetailsEntity
import com.example.cqrsread.domain.feed.persistence.entity.FeedsEntity
import org.springframework.stereotype.Component

@Component
class FeedsMapper {
    suspend fun toDomain(entity: FeedsEntity): Feeds {
        return Feeds(
            id = entity.id,
            title = entity.title,
            createdAt = entity.createdAt,
            username = entity.username,
        )
    }

    suspend fun toEntity(domain: Feeds): FeedsEntity {
        return FeedsEntity(
            id = domain.id,
            title = domain.title,
            createdAt = domain.createdAt,
            username = domain.username,
        )
    }
}

@Component
class FeedDetailsMapper {
    suspend fun toDomain(entity: FeedDetailsEntity): FeedDetails {
        return FeedDetails(
            id = entity.id,
            title = entity.title,
            content = entity.content,
            username = entity.username,
            createdAt = entity.createdAt,
        )
    }

    suspend fun toEntity(domain: FeedDetails): FeedDetailsEntity {
        return FeedDetailsEntity(
            id = domain.id,
            title = domain.title,
            content = domain.content,
            username = domain.username,
            createdAt = domain.createdAt,
        )
    }
}